include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -103,6 +103,12 @@ arena_prof_info_set(tsd_t *tsd, edata_t *edata, prof_tctx_t *tctx) {
 	large_prof_info_set(edata, tctx);
 }
 
+JEMALLOC_ALWAYS_INLINE bool
+arena_may_force_decay(arena_t *arena) {
+	return !(arena_dirty_decay_ms_get(arena) == -1
```

Comment:
Is this the right condition? The arena may force decay if either one is non-negative, not only if both are non-negative.

Reply:
Actually, if either is -1, it won't force_purge. That means the extent cache will queue at either dirty or muzzy states, and never make to retained / unmapped state?